### DISCUSSION TITLE
Responsive breeze images

### DISCUSSION THREAD

**Discussion by pk-pankaj** [breezefront/community#212] [1. Help]
Hi @vovayatsyuk 

Q1. How we can disable responsive images for particular id like "category_page_list" while we have enabled from theme config to use on PDP?

Q2. and how responsive image applied for the particular sizes, see below as I have set for the id:

```xml
<image id="category_page_list" type="small_image">
    <width>329</width>
    <height>329</height>
</image>
<image id="category_page_list-srcset-1" type="small_image">
    <width>260</width>
    <height>260</height>
</image>
<image id="category_page_list-srcset-2" type="small_image">
    <width>380</width>
    <height>380</height>
</image>
```

```xml
 <var name="category_page_list-1column"><![CDATA[
    (min-width: 1024px) 380px,
    (min-width: 768px) 260px,
    (min-width: 320px) 329px,
    calc((100vw - 1.75rem) * 0.28)
]]></var>
```

It's loading 380 size images for all screens

Thanks.

**Comment by vovayatsyuk** [comment]
> How we can disable responsive images for particular id like "category_page_list" while we have enabled from theme config to use on PDP?

You can use the same image size in all category_page_list srcsets, this may work:

```xml
<image id="category_page_list" type="small_image">
    <width>329</width>
    <height>329</height>
</image>
<image id="category_page_list-srcset-1" type="small_image">
    <width>329</width>
    <height>329</height>
</image>
<image id="category_page_list-srcset-2" type="small_image">
    <width>329</width>
    <height>329</height>
</image>
<image id="category_page_list-srcset-3" type="small_image">
    <width>329</width>
    <height>329</height>
</image>
...
```

> how responsive image applied for the particular sizes

Breeze generates the following image markup:

```html
<img class="hover-image product-image-photo" width="353" height="439" alt="Erika Running Short"
  srcset="wsh12-green_alt1_1.jpg 353w,wsh12-green_alt1_1.jpg 482w,wsh12-green_alt1_1.jpg 537w,wsh12-green_alt1_1.jpg 706w" 
  sizes="(min-width: 1360px) 324px, (min-width: 1024px) calc((100vw - 4rem) / 4), (min-width: 640px) calc((100vw - 3rem) / 3), calc((100vw - 2rem) / 2)" src="wsh12-green_alt1_1.jpg">
```

As you can see we declare a set of images of particular dimensions (srcset attribute). We also declare the rules that should be used to choose the image (sizes attribute).
 
Everything else is handled by the browser. 

For example, when the viewport size is 650px, the browser will:

1. Find the first suitable rule: `(min-width: 640px) calc((100vw - 3rem) / 3)`. 
2. Since I used the `calc` expression as the image width, the browser will calculate it: 650-3rem/3=201px.
3. Convert image size according to DPI: 201*2=402px
4. Take the most suitable image from srcset.

**Reply by pk-pankaj** [reply]
Hi @vovayatsyuk 

I am still not clear on this, I have created src sets and src set rules see below:

Src sets

`<image id="category_page_grid" type="small_image">
                <width>260</width>
                <height>260</height>
                <transparency>false</transparency>
                <background>[246,246,246]</background>
            </image>
            <image id="category_page_grid-srcset-1" type="small_image">
                <width>329</width>
                <height>329</height>
                <transparency>false</transparency>
                <background>[246,246,246]</background>
            </image>
            <image id="category_page_grid-srcset-2" type="small_image">
                <width>460</width>
                <height>460</height>
                <transparency>false</transparency>
                <background>[246,246,246]</background>
            </image>`

Src ruels

` <var name="category_page_grid-1column"><![CDATA[
                (min-width: 1536px) 460px,
                (min-width: 1024) 329px,
                (min-width: 640px) 260px
            ]]></var>`

so in this case I assume the 260 size images should be loaded with min-width: 640, and 329 should be for 1024 and 460 for 1536, am I right? but it is loading 460 for all screens.

**Reply by vovayatsyuk** [reply]
What does the resulting image HTML look like? What's your DPI? Do you have a link to your site?

**Reply by pk-pankaj** [reply]
Hi @vovayatsyuk 

Actually, I am not clear how this shows the images on the front end for a particular size and how creates rules and dpi, can you please create a src set and rules for example so I can try to understand that?

my required image sizes are:

on mobile 329px
on tablet   260
and desktop 480

**Reply by vovayatsyuk** [reply]
For me, your sizes and images look fine. Please provide the link to your site.

**Reply by pk-pankaj** [reply]
this is in the local environment yet.

image source is creating like below, if help, this is always loading 460 image for all view size

`<img class="product-image-photo" srcset="http://artuk.local.com/media/catalog/product/cache/ccd8b75de3904496b5b58f96b74274e3/c/h/christ-with-his-followers.jpeg 260w,http://artuk.local.com/media/catalog/product/cache/4bc43a30b9f92566b1aad13581a4eaea/c/h/christ-with-his-followers.jpeg 329w,http://artuk.local.com/media/catalog/product/cache/01f401177b9aeefc6f3407a40cf3477a/c/h/christ-with-his-followers.jpeg 460w" sizes="(min-width: 1536px) 460px, (min-width: 1024) 329px, (min-width: 640px) 260px" src="http://artuk.local.com/media/catalog/product/cache/ccd8b75de3904496b5b58f96b74274e3/c/h/christ-with-his-followers.jpeg" width="260" height="260" alt="Christ with His Followers">`

**Reply by vovayatsyuk** [reply]
Looks good for me. If you are using a monitor with a high DPI resolution browser will show a bigger size to prevent blurry images.

Sorry, I can't help more without the link.

**Reply by pk-pankaj** [reply]
Thanks @vovayatsyuk 
will share link when this will be on the server.

**Reply by vovayatsyuk** [reply]
Just noticed that your sizes missing image for small screens (min-width: 640px doesn't match any phone). The proper way to write it is:

```
(min-width: 1536px) 460px, (min-width: 1024) 329px, 260px
```

I removed the `(min-width: 640px) 260px` part because I assume you want to load 260w images for mobile devices too. (However, since all phones have a high pixel ratio (1.8-3x), the browser will try to find a match for 260xRatio)
